src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -0,0 +1,178 @@
+            static void checkArg(string arg, string name)
+            {
+                if (String.IsNullOrEmpty(arg))
+                    throw new ArgumentException(name);
+            };
+            checkArg(scenarioName, nameof(scenarioName));
+            checkArg(appExe, nameof(appExe));
+            checkArg(traceFileName, nameof(traceFileName));
+
+            bool failed = false;
+            logger.Log($"Running {appExe} (args: \"{appArgs}\")");
+            var procHelper = new ProcessHelper()
+            {
+                ProcessWillExit = processWillExit,
+                Timeout = timeout,
+                MeasurementDelay = measurementDelay,
+                Executable = appExe,
+                Arguments = appArgs,
+                WorkingDirectory = workingDir,
+                GuiApp = guiApp
+            };
+            Util.Init();
+
+            if (warmup)
+            {
+                procHelper.Run();
+            }
+
+            string kernelTraceFile = Path.ChangeExtension(traceFileName, "perflabkernel.etl");
+            string userTraceFile = Path.ChangeExtension(traceFileName, "perflabuser.etl");
+            IParser parser = null;
+            switch (metricType)
+            {
+                case MetricType.TimeToMain:
+                    parser = new TimeToMainParser();
+                    break;
+                case MetricType.GenericStartup:
+                    parser = new GenericStartupParser();
+                    break;
+                    //case MetricType.WPF:
+                    //    parser = new WPFParser();
+                    //    break;
+            }
+            using (var kernel = new TraceEventSession(KernelTraceEventParser.KernelSessionName, kernelTraceFile))
+            {
+                parser.EnableKernelProvider(kernel);
+                using (var user = new TraceEventSession("StartupSession", userTraceFile))
+                {
+                    parser.EnableUserProviders(user);
+                    for (int i = 0; i < iterations; i++)
+                    {
+                        var result = procHelper.Run();
+                        if (result != ProcessHelper.Result.Success)
+                        {
+                            logger.Log($"failed. result: {result}");
+                            failed = true;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            if (!failed)
+            {
+                logger.Log("Parsing..");
+                TraceEventSession.Merge(new[] { kernelTraceFile, userTraceFile }, traceFileName);
+                var counters = parser.Parse(traceFileName, Path.GetFileNameWithoutExtension(appExe));
+                foreach (var counter in counters)
+                {
+                    logger.Log($"{counter.Name,-15}: {counter.Results.Average():F3} {counter.MetricName}");
```

Comment:
Are we going to add more statistics here, instead of just average?  It seems for non-micro-benchmarks more values like min, max, std ... could help prove the stability of the results.

Reply:
We can add anything we want.